src/setup/node.rs

```diff
@@ -114,13 +115,22 @@ impl Node {
             .expect("Node failed to start");
         self.child = Some(child);
 
+        // Once the node is started, fetch its addresses.
+        self.conf
+            .load_addr()
+            .await
+            .expect("Couldn't load the node's addresses.");
+
         // TODO(Rqnsom) wait for the connection to confirm.
     }
 
     /// Stops the node instance.
     pub fn stop(&mut self) -> io::Result<ChildExitCode> {
         // Cannot use 'mut self' due to the Drop impl.
 
+        self.conf.local_addr = None;
+        self.conf.rest_api_addr = None;
+
```

Comment:
do we need to "zero out" the addresses?

Reply:
Hmm, fair enough; I was thinking in terms of `Drop`, where it's without consequence, but if it's done manually, it might make a difference.